examples/knative/go/kn-go-echo/server.go

```diff
@@ -0,0 +1,29 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"log"
+
+	cloudevents "github.com/cloudevents/sdk-go/v2"
+)
+
+func main() {
+	ctx := context.Background()
+	protocol, err := cloudevents.NewHTTP()
+	if err != nil {
+		log.Fatalf("creating protocol: %s", err.Error())
+	}
+
+	client, err := cloudevents.NewClient(protocol)
+	if err != nil {
+		log.Fatalf("creating client, %v", err)
+	}
+
+	log.Printf("listening on :8080\n")
+	log.Fatalf("start receiver: %s", client.StartReceiver(ctx, receive))
+}
+
+func receive(ctx context.Context, event cloudevents.Event) {
```

Comment:
Shall we take this to the next level and also return the received event in case this example is used as part of a pipeline?

Reply:
Yes, I like that idea. I'll return the cloud event.

Reply:
@embano1, to what address should I forward the event to? I guess it's the rabbitmq event broker? Would the address be something like `https://rabbit-broker-ingress.default.svc:80`?

Reply:
You simply return the event like here https://github.com/cloudevents/sdk-go/blob/main/samples/http/responder/main.go#L50

`receiver` is allowed to have different (pre-defined) signatures

Reply:
Nice--thanks for finding that example. I went ahead and implemented responding with the event.

Reply:
I had to add a fix because by returning the same event, it caused an infinite event loop. I changed the event type and added a trigger filter so event types from `kn-go-echo` did not get passed along to `kn-go-echo`.